src/session.c

```diff
@@ -264,7 +264,7 @@ void p11prov_free_slots(P11PROV_SLOTS_CTX *sctx)
             OPENSSL_clear_free(sctx->slots[i]->bad_pin,
                                strlen(sctx->slots[i]->bad_pin));
         }
-        OPENSSL_free(sctx->slots[i]->login_info);
+        free(sctx->slots[i]->login_info);
```

Comment:
That memory is [allocated](https://github.com/latchset/pkcs11-provider/blob/5c029461e4f886f821d7333378c70fd3bb0ca879/src/session.c#L210) with `asprintf`, which should be using system malloc rather than `OPENSSL_malloc`. We could replace that part with `asprintf` → `OPENSSL_strdup` → `free`, but I thought it would be too redundant.

Reply:
Thanks!

Reply:
@beldmit is there an asprintf-like function in OpenSSL? I couldn't find one and I am not too happy about using asprintf as it requires _GNU_SOURCE and as you can see from this issue, is prone to mistakes in mixing free/openssl_free :/

Reply:
Don't think so. You can do smth around BIO_printf, I think